spec/01-unit/08-router_spec.lua

```diff
@@ -1156,6 +1156,33 @@ describe("Router", function()
         assert.same(nil, match_t.matches.uri_captures)
       end)
 
+      it("matches a [wildcard host + port] even if a [wildcard host] matched", function()
+        local use_case = {
+          {
+            service = service,
+            route = {
+              hosts = { "route.*" },
+            },
+          },
+          {
+            service = service,
+            route = {
+              hosts = { "route.*:123" },
+            },
+          },
+        }
+
+        local router = assert(Router.new(use_case))
+
+        local match_t = router.select("GET", "/", "route.org:123")
+        assert.truthy(match_t)
+        assert.equal(use_case[2].route, match_t.route)
+        assert.same("route.*:123", match_t.matches.host)
+        assert.same(nil, match_t.matches.method)
+        assert.same(nil, match_t.matches.uri)
+        assert.same(nil, match_t.matches.uri_captures)
+      end)
```

Comment:
This was covered in a few test case, but a bit too indirectly, and I'm afraid the test may be updated in the future and not cover the case, so here is an isolated test case specifically for the submatching priority.